### ISSUE TITLE
Add RequiresMountsFor for gvisor tap device systemd unit file

### ISSUE THREAD

**Issue by praveenkumar** [code-ready/snc#436]
During 4.8 testing when trying to poweroff the system instead of graceful shutdown because of https://bugzilla.redhat.com/show_bug.cgi?id=1965992  then `gvisor-tap-device` fail to start and as per podman doc http://docs.podman.io/en/latest/markdown/podman-generate-systemd.1.html the generate systemd unit file should have the `RequiresMountsFor` so it can wait till that mount point before starting the service.

```
open /var/lib/containers/storage/libpod/bolt_state.db: no such file or directory
```

Also better to use `--new` option to generate the shareable unit files.

**Comment by praveenkumar** [comment]
`RequiresMountsFor` is part of the podman 3.1.x and right now in the 4.8 it have 3.0.x podman version. So this will fix as soon as new version of podman is available in coreOS.

**Comment by praveenkumar** [comment]
With latest version of podman it is automatic fixed (3.4.x) so closing it.
